app/src/main/java/woowacourse/movie/view/reservation/seat/SeatSelectionPresenter.kt

```
@@ -0,0 +1,55 @@
+    override fun selectSeat(seat: Seat) {
+        try {
+            reservationInfo?.updateSeats(seat)
+            view.updateSeatSelection(seat, seat.isSelected)
+            updateScreen()
+        } catch (e: IllegalArgumentException) {
+            view.showError(e.message ?: "좌석 선택 오류")
+        }
+    }
+
+    private fun updateScreen() {
+        view.showTotalPrice(reservationInfo?.totalPrice() ?: throw IllegalArgumentException())
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)